test/functional/shada/shada_spec.lua

```diff
@@ -262,4 +262,14 @@ describe('ShaDa support code', function()
        exc_exec('wshada'))
     meths.set_option('shada', '')
   end)
+
+  it('does not store unlisted buffer', function()
+    local fname = funcs.getcwd() .. '/file'
+    meths.set_var('__fname', fname)
+    nvim_command('silent! edit `=__fname`')
+    curbufmeths.set_option('buflisted', false)
+    nvim_command('wshada! ' .. shada_fname)
+
+    eq({}, find_file(fname))
```

Comment:
to avoid false negatives (tests that pass but shouldn't), it helps to check the reverse, so e.g. after this line add a small sanity check:


    curbufmeths.set_option('buflisted', true)
    nvim_command('wshada! ' .. shada_fname)
    eq(..., find_file(fname))

Reply:
Makes sense.

e12dee2440177cb4930ff3e8912aeb74860d8fd8